src/openjd/adaptor_runtime/_entrypoint.py

```diff
@@ -123,13 +159,21 @@ class EntryPoint:
     The main entry point of the adaptor runtime.
     """
 
+    on_bootstrap_complete: MutableSet[Callable[[], None]]
```

Comment:
I'm not sure that I like the list of callbacks pattern established here and above in the Background Runner. Right now we have a 
- Set of callbacks in the `_LogConfig` which we add a single callback to in order to disconnect the file log handlers, but removes itself from the set of callbacks when called. 
- A set of callbacks in the BackgroundRunner which is only invoked when the connection file is written, which we added in order to provide the aforementioned callback

It looks like the goal is to disconnect the bootstrap logger after the connection file is written. I appreciate the forward thinking but I think in this case I'd prefer biasing towards keeping things simple and just having the background runner call a function directly which disconnects the bootstrap logging. It would allow us to keep the flow between these components more directly readable

Reply:
Yeah, the idea is to make it generic so we do not need to refactor it in the future if we ever need to subscribe more actions to when the bootstrap is complete. Here I am trying to follow a really basic version of "events" in other languages, like C#.